src/Distribution/Server/Framework/HappstackUtils.hs

```diff
@@ -172,14 +172,14 @@ enableGZip compressParams modifyETag = do
   where
     gzipFilter :: Response -> Response
     gzipFilter r@SendFile{} = r -- Leave files alone
-    gzipFilter r@Response{} =
+    gzipFilter Response{..} =
         chunked
       . setHeader "Content-Encoding" "gzip"
       . setHeader "Vary" "Accept-Encoding"
       . removeResponseHeader "Content-Length"
       . removeResponseHeader "Content-MD5"
       . alterResponseHeader "ETag" (map modifyETag)
-      $ r { rsBody = GZip.compressWith compressParams $ rsBody r }
+      $ Response { rsBody = GZip.compressWith compressParams rsBody, .. }
```

Comment:
I find this version (with the magic dots `..` in the record update) harder to comprehend.  I consider this a fringe feature of GHC Haskell (I am seeing it for the first time now).
There might be some risks of unintentionally introducing bugs by shadowing one of the record fields introduced by the `..` pattern which is then picked up by the `..` update, is there?

I think that in the contrary we should strive to get rid of all the `..` magic and be more explicit about what is in scope.

So, I'd rather not change code to work around an incorrect GHC warning mechanism.  Rather switch off the warning.